build-support/bin/generate_github_workflows.py

```diff
@@ -545,8 +546,38 @@ def upload_log_artifacts(self, name: str) -> Step:
             "if": "always()",
             "continue-on-error": True,
             "with": {
-                "name": f"pants-log-{name.replace('/', '_')}-{self.platform_name()}",
-                "path": ".pants.d/pants.log",
+                "name": f"logs-{name.replace('/', '_')}-{self.platform_name()}",
+                "path": ".pants.d/*.log",
+            },
+        }
+
+    def upload_test_reports(self) -> Step:
+        date_str = datetime.now(timezone.utc).date().isoformat()
+        # The path doesn't include job ID, as we want to aggregate test reports across all
```

Comment:
Hmm I guess it comes down to whether we want files to be overwritten in that case or not? It shouldn't happen with our current setup, since we don't run the same tests on the same platform in different jobs. If we don't want them overwritten then yeah, adding the job ID is wise.

Reply:
On balance, I'd personally lean towards avoiding overwrites:

- If we avoid overwrites and want latest-wins behaviour, I suspect we can implement it at read-time by referring to S3 last-modified metadata.
- If we implicitly overwrite and want every-file behaviour, I think it's impossible. I'd also suspect it's hard to notice if we accidentally violated the assumption that per-job files are unique and started overwriting files, because analysis might still give plausible numbers on the subset that survives.

An alternative/addition to job ID would be to turn on S3 bucket versioning, but that feels somewhat over the top.